### PR TITLE
[modbus] Modbus transformations: cascaded/chained transformations and new-style transformation string

### LINKED ISSUE #9944: [modbus] Modbus transformations should support cascaded transformations with ∩
Similar to HTTP and serial binding, modbus should support cascading/chaining transformations with ∩.

(will provide PR)

### LINKED ISSUE #9946: [modbus] Modbus binding should support new style transformation string SERVICE:ARG
Modbus binding transformation currently supports only "old-style" transformation string `SERVICE(ARG)`. Similar to other bindings, we should also support `SERVICE:ARG`.

### PR DESCRIPTION
**Pull request by ssalonen**
Resolves #9944
Resolves #9946

### CODE REVIEW

**Comment by ssalonen** [comment]
Rebuilding, let's see if Jenkins would pass then.

**Comment by ssalonen** [comment]
Can someone find why the jenkins fails? I cannot find proper error message and everything seems to be interleaved in the logs?

**Comment by ssalonen** [comment]
Rebased to get https://github.com/openhab/openhab-addons/pull/9702

Locally failing with:
```
$ mvn -pl itests/org.openhab.binding.modbus.tests,bundles/org.openhab.binding.modbus install -DwithResolver=true
...
[ERROR] Failed to execute goal biz.aQute.bnd:bnd-resolver-maven-plugin:5.2.0:resolve (default) on project org.openhab.binding.modbus.tests: Unable to resolve <<INITIAL>>: missing requirement osgi.identity;filter:='(osgi.identity=org.openhab.binding.modbus.tests)' [caused by: Unable to resolve org.openhab.binding.modbus.tests version=3.1.0.202101261800: missing requirement osgi.wiring.package;filter:='(&(osgi.wiring.package=org.junit.jupiter.api)(version>=5.7.0)(!(version>=6.0.0)))';osgi.wiring.package='org.junit.jupiter.api'] -> [Help 1]
```

Let's see what Jenkins thinks

**Comment by ssalonen** [comment]
```
[ERROR] Failed to execute goal org.openhab.tools.sat:sat-plugin:0.10.0:spotbugs (sat-all) on project org.openhab.automation.jythonscripting: Unable to execute mojo: Execution null of goal com.github.spotbugs:spotbugs-maven-plugin:4.0.0:spotbugs failed: Timeout: killed the sub-process -> [Help 1]
```

**Comment by ssalonen** [comment]
Apache commons dependency removed from Transformation class in https://github.com/openhab/openhab-addons/pull/9945/commits/8f0042506414ab1250ad341c82ed2e1f9dbe046f. Other references removed in https://github.com/openhab/openhab-addons/pull/10002

File: bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/ValueTransformation.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TypeParser;
+import org.osgi.framework.BundleContext;
+
+/**
+ * Interface for Transformation
+ *
+ * @author Sami Salonen
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/ValueTransformation.java:26]:
```suggestion
 * @author Sami Salonen - Initial contribution
```

**Review by Hilbrand — COMMENTED** [review]
Just one small comment and a rebase is needed.

**Review by Hilbrand — APPROVED** [review]
LGTM

**Pull request merged**